tests/e2e/testsLibrary/LanguageServerTests.ts

```diff
@@ -75,21 +75,42 @@ export class LanguageServerTests {
         });
     }
 
-    public codeNavigation(openedFile: string, line: number, char: number, codeNavigationClassName: string, timeout : number = TimeoutConstants.TS_EDITOR_TAB_INTERACTION_TIMEOUT) {
-        test('Codenavigation', async () => {
+    public goToDefinition(openedFile: string, line: number, char: number, codeNavigationClassName: string, timeout : number = TimeoutConstants.TS_EDITOR_TAB_INTERACTION_TIMEOUT) {
+        test('Go to Definition', async () => {
+            // adding retry to fix https://github.com/eclipse/che/issues/17411
+            try {
+                await this.editor.moveCursorToLineAndChar(openedFile, line, char);
+                await this.editor.performKeyCombination(openedFile, Key.chord(Key.CONTROL, Key.F11));
+                await this.editor.waitEditorAvailable(codeNavigationClassName, timeout);
+            } catch (err) {
+                if (err instanceof error.TimeoutError) {
+                    Logger.warn('Code navigation (definition) didn\'t work. Trying again.');
+                    await this.editor.moveCursorToLineAndChar(openedFile, line, char);
+                    await this.editor.performKeyCombination(openedFile, Key.chord(Key.CONTROL, Key.F11));
+                    await this.editor.waitEditorAvailable(codeNavigationClassName, timeout);
+                } else {
+                    Logger.error('Code navigation (definition) didn\'t work even after retrying.');
```

Comment:
If it will fail during retrying, it will not get to the `else` statement AFAIU. There is just one `try`.

Reply:
fixed.